### ISSUE TITLE
Top nav navigation group `aria-expanded` value not always correct

### ISSUE THREAD

**Issue by GCHQ-Developer-847** [mi6/ic-ui-kit#3392]
## Summary of the bug 
Navigation groups on a top navigation component don't always get read out as "expanded" or "collapsed" correctly when using a screen reader.

## 🪜 How to reproduce 
1. Go to this [top navigation with nav groups StackBlitz](https://stackblitz.com/edit/n6uvzfsk?file=src%2Fapp.tsx&hideExplorer=1) (on a desktop / large screen size).
2. While using a screen reader, focus on one of the navigation groups.
3. See that it gets read out as "expanded" when it is actually collapsed.

It does sometimes get read out correctly e.g. open one of the groups, tab onto a child nav item, and tab back onto the group. But do this again and it says it is expanded, even though it isn't.

Try this on the side menu displayed on a mobile / small screen size as well and there is the same issue (except it never seems to say that the group is collapsed).

## 🖥 📱 Device 
Tested using VoiceOver on Chrome.

## 🧐 Expected behaviour 
The aria-expanded value should be changed and read out correctly when the navigation group is expanded or collapsed.

## Additional info 
The side nav navigation groups seem to work correctly; it is important when working on this ticket to ensure the side nav aria-expanded behaviour is not affected / continues to work.
